subprojects/groovy-sql/src/main/java/groovy/sql/BatchingStatementWrapper.java

```diff
@@ -83,8 +83,15 @@ public void clearBatch() throws SQLException {
     }
 
     public int[] executeBatch() throws SQLException {
-        int[] lastResult = delegate.executeBatch();
-        processResult(lastResult);
+        if (batchCount > 0) {
+            int[] lastResult = delegate.executeBatch();
+            processResult(lastResult);
+        }
+        else if (results.isEmpty()) {
+            log.warning("Nothing has been added to batch");
+            // we let the JDBC provider decide how to handle an empty batch execution
+            return delegate.executeBatch();
+        }
```

Comment:
This "else if" is not really elegant for such a particular case...
I could have done `if (batchCount > 0 || results.isEmpty())` in the first condition, but I thought it made the intention unreadable.